src/components/Components/ComponentDetails/tabs/ComponentNudgesDependencies.tsx

```diff
@@ -0,0 +1,202 @@
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import { ExpandableSection, Radio, Tooltip } from '@patternfly/react-core';
+import { OutlinedQuestionCircleIcon } from '@patternfly/react-icons/dist/js/icons/outlined-question-circle-icon';
+import { ComponentKind, NudgeSpecs } from '../../../../types';
+import { useWorkspaceInfo } from '../../../../utils/workspace-context-utils';
+
+import './ComponentNudgesDependencies.scss';
+
+type ComponentBuildSettingsProps = {
+  component: ComponentKind;
+};
+
+enum NudgeRadios {
+  NUDGES = 'nudges',
+  NUDGED_BY = 'isNudgedBy',
+}
+
+const ComponentBuildSettings: React.FC<React.PropsWithChildren<ComponentBuildSettingsProps>> = ({
+  component,
+}) => {
+  const [radioChecked, setRadioChecked] = React.useState<string>(NudgeRadios.NUDGES);
+  const [expanded, setExpanded] = React.useState<boolean>(true);
+  const { workspace } = useWorkspaceInfo();
+  const nudges = component?.spec && component.spec[NudgeSpecs.NUDGES];
```

Comment:
spec will always be present on this resource. Also use Optional chaining instead of `&&`

```
component.spec?.[NudgeSpecs.Nudges]
```